packages/data/src/components/use-select/test/index.js

```diff
@@ -132,5 +132,89 @@ describe( 'useSelect', () => {
 			children: 'bar',
 		} );
 	} );
+	describe( 'rerenders as expected with various mapSelect return ' +
+		'types', () => {
+		const getComponent = ( mapSelectSpy, dependencyKey ) => ( props ) => {
```

Comment:
These don't really need to be mocks and using normal functions would make the test easier to read.

We should only use mocks when we need to read from the `.mock` property. E.g. to introspect on what parameters the function was called with.

Reply:
I converted two of the mocks that were unnecessary in 511b504.  However, the others still had utility for both spying on the calls and changing the return value.